types/fs-extra/index.d.ts

```diff
@@ -232,9 +235,9 @@ export function write(fd: number, data: any, offset: number, encoding: string, c
 export function write(fd: number, buffer: Buffer, offset: number, length: number, position?: number | null): Promise<[number, Buffer]>;
 export function write(fd: number, data: any, offset: number, encoding?: string): Promise<[number, string]>;
 
-export function writeFile(filename: string, data: any, callback: (err: NodeJS.ErrnoException) => void): void;
-export function writeFile(filename: string, data: any, options: { encoding?: string; mode?: number; flag?: string; }, callback?: (err: NodeJS.ErrnoException) => void): void;
-export function writeFile(filename: string, data: any, options?: { encoding?: string; mode?: number; flag?: string; }): Promise<void>;
+export function writeFile(filename: string | number, data: any, callback: (err: NodeJS.ErrnoException) => void): void;
```

Comment:
`file` can be `string | Buffer | number` according to the docs: https://nodejs.org/dist/latest-v7.x/docs/api/fs.html#fs_fs_writefile_file_data_options_callback

Also, let's call this`file` rather than `filename` to indicate the flexibility.